x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/optimizer/TestLocalPhysicalPlanOptimizer.java

```diff
@@ -13,12 +13,19 @@
 
 public class TestLocalPhysicalPlanOptimizer extends LocalPhysicalPlanOptimizer {
 
+    private final boolean esRules;
```

Comment:
nit: the variable name used in the `rules` method is clearer (needed to look this up to understand what's going on), consider renaming:
```suggestion
    private final boolean optimizeForEsSource;
```